### PR TITLE
RBAC: fix access control check in both new navivation and new search view

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
I noticed that when using the new navigation flag the access control permissions was not checked, this was also true for the new search view in front-end.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/index.go [hunk 1]
```diff
@@ -492,19 +497,26 @@ func (hs *HTTPServer) buildDashboardNavLinks(c *models.ReqContext, hasEditPerm b
 		dashboardChildNavs = append(dashboardChildNavs, &dtos.NavLink{
 			Text: "Divider", Divider: true, Id: "divider", HideFromTabs: true,
 		})
-		dashboardChildNavs = append(dashboardChildNavs, &dtos.NavLink{
-			Text: "New dashboard", Icon: "plus", Url: hs.Cfg.AppSubURL + "/dashboard/new", HideFromTabs: true, Id: "new-dashboard", ShowIconInNavbar: true,
-		})
-		if c.OrgRole == models.ROLE_ADMIN || c.OrgRole == models.ROLE_EDITOR {
+
+		if hasAccess(hasEditPermInAnyFolder, ac.EvalPermission(dashboards.ActionDashboardsCreate)) {
```
Comment (IevaVasiljeva) [inline pkg/api/index.go:501]: nit: could this conditional could be combined with the dashboard import one on line 514?

**Review by IevaVasiljeva — APPROVED** [review]
LGTM, good catch!

File: pkg/api/index.go (hunk 1 above)
Comment (gamab) [reply pkg/api/index.go:501]: I think it would affect the order in which the NavLinks are displayed.

**Review by gamab — APPROVED** [review]
Good catch!

File: public/app/features/search/components/ManageDashboards.tsx [hunk 2]
```diff
@@ -101,7 +101,14 @@ export const ManageDashboards: FC<Props> = memo(({ folder }) => {
         <div className="gf-form gf-form--grow m-r-2">
           <FilterInput value={query.query} onChange={onQueryChange} placeholder={'Search dashboards by name'} />
         </div>
-        <DashboardActions isEditor={isEditor} canEdit={hasEditPermissionInFolders || canSave} folderId={folderId} />
+        <DashboardActions
+          folderId={folderId}
+          canCreateFolders={contextSrv.hasAccess(AccessControlAction.FoldersCreate, isEditor)}
+          canCreateDashboards={contextSrv.hasAccess(
+            AccessControlAction.DashboardsCreate,
```
[the remark this answers is not in the document]
Comment (kalleep) [reply public/app/features/search/components/ManageDashboards.tsx:108]: We dont have access control meta in folder atm https://github.com/grafana/grafana-enterprise/issues/3401

**Pull request merged**